lib/batch-request.js

```
@@ -58,10 +61,28 @@ var batchRequest = function(params) {
                         r.headers[headerK] = forwardValue;
                     }
                 });
+
+                r.encoding = null;
                 promises[key] = request(r).spread(function(response, body) {
+                    var encoding = 'utf8';
+                    if (params.smartDecode) {
+                        try {
+                            var contentTypeInfo = contentType.parse(response.headers['content-type']);
+                            encoding = contentTypeInfo.parameters.charset;
+                            if (!Buffer.isEncoding(encoding)) {
+                                encoding = mime.charset(contentTypeInfo.type);
+                            }
+                            if (!Buffer.isEncoding(encoding)) {
+                                encoding = 'base64';
+                            }
+                        } catch (err) {
+                            // Could not determine charset
```

We should handle this error, at least logging something rather than silently failing